xgi/communities/spectral.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from scipy.sparse.linalg import eigsh
+
+from ..core import Hypergraph
+from ..linalg.laplacian_matrix import normalized_hypergraph_laplacian
+
+from ..exception import XGIError
+
+__all__ = [
+    "spectral_clustering",
+]
+
+MAX_ITERATIONS = 10_000
+
+
+def spectral_clustering(H, k=None):
+    """Cluster into k-many groups using spectral techniques.
+
+    Compute a spectral clustering according to the heuristic suggested in [1].
+
+    Parameters
+    ----------
+    H : Hypergraph
+        Hypergraph
+    k : int, optional
+        Number of clusters to find. If unspecified, computes spectral gap.
+
+    Returns
+    -------
+    dict
+        A dictionary mapping node ids to their clusters. Clusters begin at 0.
+
+    Raises
+    ------
+    XGIError
+        If more groups are specified than nodes in the hypergraph.
+
+
+    References
+    ----------
+    .. [1] Zhou, D., Huang, J., & Schölkopf, B. (2006).
+        Learning with Hypergraphs: Clustering, Classification, and Embedding
+        Advances in Neural Information Processing Systems.
+
+    """
+    if k is None:
+        raise NotImplementedError(
+            "Choosing a number of clusters organically is currently unsupported. Please specify an integer value for paramater 'k'!"
+        )
+    else:
+        if k > H.num_nodes:
+            raise XGIError(
+                "The number of desired clusters cannot exceed the number of nodes!"
+            )
+
+    # Compute normalize Laplacian and its spectra
+    L, rowdict = normalized_hypergraph_laplacian(H, index=True)
+    evals, eigs = eigsh(L, k=k, which="SA")
+
+    # Form metric space representation
+    X = np.array(eigs)
+    print(X.shape, X)
+
+    # Apply k-means clustering
+    _clusters = _kmeans(X, k)
+
+    # Remap to node ids
+    clusters = {rowdict[id]: cluster for id, cluster in _clusters.items()}
+
+    return clusters
+
+
+def _kmeans(X, k, seed=37):
```

Comment:
Add docstring. Also, is there a reason to specify the seed by default instead of `seed=None` like in other functions?